node/src/actors/chain_manager/handlers.rs

```diff
@@ -1092,9 +1133,10 @@ impl Handler<PeersBeacons> for ChainManager {
 
                         // If we are synced, it does not matter what blocks have been consolidated
                         // by our outbound peers, we will stay synced until the next superblock
-                        // vote
+                        // vote. We will remove those that are different from the last consensus or
+                        // peers that has a block different to us
 
-                        Ok(peers_with_no_beacon)
```

Comment:
This change can have some big implications, any context on why it is needed?

Reply:
When we are synced and there is no consensus, we select as peers_to_unregister those that are different to us. But later we only use the peers_with_no_beacon. This change is to use the first part of the code.

Reply:
When we are synced and there is no consensus, all of the other peers may be different, so by dropping all the peers we will have worse connectivity... Ideally we should only drop the "bad" peers but we have no way to define "bad".